sdk/iot/pnp/inc/az_iot_pnp_client.h

```diff
@@ -0,0 +1,332 @@
+ * initialize the buffer.
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[in] component_name An #az_span specifying the component name to publish telemetry on.
+ * @param[in] content_type An #az_span specifying the content type of the message payload.
+ * @param[in] encoding_type An #az_span specifying the encoding type of the message payload.
+ * @param[in] options A reference to an #az_iot_pnp_client_telemetry_options structure. Can be NULL.
+ * @param[in] mqtt_topic An empty #az_span with sufficient capacity to hold the MQTT topic.
+ * @param[out] out_mqtt_topic The output #az_span containing the MQTT topic.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_pnp_client_telemetry_publish_topic_get(
+    az_iot_pnp_client const* client,
+    az_span component_name,
+    az_span content_type,
+    az_span encoding_type,
+    az_iot_pnp_client_telemetry_options const* options,
+    az_span mqtt_topic,
+    az_span* out_mqtt_topic);
+
+/*
+ *
+ * PnP Command APIs
+ *
+ */
+
+/**
+ * @brief Azure IoT PnP command subscription options.
+ *
+ */
+typedef struct az_iot_pnp_client_command_subscription_options
+{
+  struct
+  {
+    int _not_used;
+  } _internal;
+} az_iot_pnp_client_command_subscription_options;
+
+/**
+ * @brief Gets the default Azure IoT PnP command subscription options.
+ * @details Call this to obtain an initialized #az_iot_pnp_client_command_subscription_options
+ * structure that can be afterwards modified and passed to
+ * #az_iot_pnp_client_commands_subscribe_topic_filter_get.
+ *
+ * @return #az_iot_pnp_client_command_subscription_options.
+ */
+AZ_NODISCARD az_iot_pnp_client_command_subscription_options az_iot_pnp_client_command_subscription_options_default();
+
+/**
+ * @brief Gets the MQTT topic filter to subscribe to PnP commands.
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[in] options A reference to an #az_iot_pnp_client_command_subscription_options structure.
+ * Can be NULL.
+ * @param[in] mqtt_topic_filter An empty #az_span with sufficient capacity to hold the MQTT topic
+ *                              filter.
+ * @param[out] out_mqtt_topic_filter The output #az_span containing the MQTT topic filter.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_pnp_client_commands_subscribe_topic_filter_get(
+    az_iot_pnp_client const* client,
+    az_iot_pnp_client_command_subscription_options const* options,
+    az_span mqtt_topic_filter,
+    az_span* out_mqtt_topic_filter);
+
+/**
+ * @brief A command request received from the PnP service.
+ *
+ */
+typedef struct az_iot_pnp_client_command_request
+{
+  az_span request_id; /**< The request id.
+                       * @note The application must match the method request and method response. */
+  az_span component; /**< The component the command is targeted towards. */
+  az_span command_name; /**< The name of command to invoke */
```

Comment:
nit: just to maintain sentence structure of other docs, add `the` in front of command

Reply:
fixed.